src/Development/IDE/Types/Options.hs

```diff
@@ -23,12 +24,22 @@ import           GhcPlugins                     as GHC hiding (fst3, (<>))
 import qualified Language.Haskell.LSP.Types.Capabilities as LSP
 import qualified Data.Text as T
 import Development.IDE.Types.Diagnostics
+import Control.DeepSeq (NFData(..))
+
+data IdeGhcSession = IdeGhcSession
+  { loadSessionFun :: FilePath -> IO (IdeResult HscEnvEq, [FilePath])
+  -- ^ Returns the Ghc session and the cradle dependencies
+  , sessionVersion :: !Int
```

Comment:
A comment on the semantics here would be great. It looks like you only use this as a shake key so the only requirement is that you never reuse a version but it’s probably a good choice to require it to be strictly increasing (but not necessarily in steps of 1).

Reply:
Added